### ISSUE TITLE
SQL injection vulnerability

### ISSUE THREAD

**Issue by wangteji** [panorama-ed/order_as_specified#21]
The `order_as_specified` method provided by this gem is vulnerable to SQL injection attacks. 

Consider the following example.
```ruby
  User.create(:secret => "foo")
  Job.create(:location => "boston")
  
  order_preference = "') jobs.location FROM jobs UNION SELECT users.secret FROM users --"
  Job.order_as_specified(distinct_on: true, location: [order_preference]).to_a.first
  => #<Job id: nil, location: "foo">
```

In the above example, I am able to replace the jobs.location field with values from users.secret. This is obviously very bad. It works because when the `:distinct_on` option is passed, the `order_as_specified` method inserts inputs into the SELECT clause without sanitizing them. Similar attacks could happen even when `:distinct_on` is not used, as `order_as_specified` also uses unsanitized inputs in the ORDER BY clause. Although the ORDER BY clause is harder to exploit than SELECT, it can still happen. See http://rails-sqli.org for some examples.

I have added a pull request to attempt to address this issue: https://github.com/panorama-ed/order_as_specified/pull/20

**Comment by JacobEvelyn** [comment]
Thanks so much for all the work here! I'll take a look tomorrow.

**Comment by JacobEvelyn** [comment]
This has been merged!

**Comment by JacobEvelyn** [comment]
This fix is now out in version 1.2. Thanks again!

**Comment by wangteji** [comment]
Awesome!
